graphdatascience/session/aura_api_responses.py

```diff
@@ -9,13 +9,15 @@
 
 from pandas import Timedelta
 
+from .session_sizes import SessionMemory
+
 
 @dataclass(repr=True, frozen=True)
 class SessionDetails:
     id: str
     name: str
     instance_id: str
-    memory: str
+    memory: SessionMemory
```

Comment:
we were not parsing this by choice. Should have put a comment.
If we introduce a new Session size option in Aura, we dont want to break client when listing sessions.

I get the point for get_or_create comparison, but for simple list it would be not ideal IMO.

Reply:
I added another commit that uses a dedicated class to represent the memory. We can keep the enum to make the size selection easier. In Java i would use constant static instances of the class instead but it does not seem like this exists in Python, so maybe there is still a better way than the enum.

Reply:
I have an HTTP call which creates 4GB instances